components/eventing-controller/controllers/subscriptionv2/eventmesh/test/reconciler_integration_test.go

```diff
@@ -0,0 +1,457 @@
+						fmt.Sprintf("%s0", reconcilertesting.OrderCreatedV1EventNotClean),
+					}),
+					reconcilertesting.WithWebhookAuthForBEB(),
+					reconcilertesting.WithSinkURL(reconcilertesting.ValidSinkURL(namespace, "test")),
+				)
+			},
+			wantSubscriptionMatchers: gomega.And(
+				reconcilertesting.HaveCondition(eventingv1alpha2.MakeCondition(
+					eventingv1alpha2.ConditionSubscriptionActive,
+					eventingv1alpha2.ConditionReasonSubscriptionActive,
+					corev1.ConditionTrue, "")),
+				reconcilertesting.HaveCleanEventTypes([]eventingv1alpha2.EventType{
+					{
+						OriginalType: fmt.Sprintf("%s0", reconcilertesting.OrderCreatedV1EventNotClean),
+						CleanType:    fmt.Sprintf("%s0", reconcilertesting.OrderCreatedV1Event),
+					}, {
+						OriginalType: fmt.Sprintf("%s1", reconcilertesting.OrderCreatedV1EventNotClean),
+						CleanType:    fmt.Sprintf("%s1", reconcilertesting.OrderCreatedV1Event),
+					},
+				}),
+			),
+			wantUpdateSubscriptionMatchers: gomega.And(
+				reconcilertesting.HaveCondition(eventingv1alpha2.MakeCondition(
+					eventingv1alpha2.ConditionSubscriptionActive,
+					eventingv1alpha2.ConditionReasonSubscriptionActive,
+					corev1.ConditionTrue, "")),
+				reconcilertesting.HaveCleanEventTypes([]eventingv1alpha2.EventType{
+					{
+						OriginalType: fmt.Sprintf("%s0", reconcilertesting.OrderCreatedV1EventNotClean),
+						CleanType:    fmt.Sprintf("%s0", reconcilertesting.OrderCreatedV1Event),
+					},
+				}),
+			),
+		},
+	}
+
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			g := gomega.NewGomegaWithT(t)
+			ctx := context.Background()
+
+			// create unique namespace for this test run
+			testNamespace := getTestNamespace()
+			ensureNamespaceCreated(ctx, t, testNamespace)
+
+			// update namespace information in given test assets
+			givenSubscription := tc.givenSubscriptionFunc(testNamespace)
+			givenUpdateSubscription := tc.givenUpdateSubscriptionFunc(testNamespace)
+
+			// create a subscriber service
+			subscriberSvc := reconcilertesting.NewSubscriberSvc(givenSubscription.Name, testNamespace)
+			ensureK8sResourceCreated(ctx, t, subscriberSvc)
+
+			// create subscription
+			ensureK8sResourceCreated(ctx, t, givenSubscription)
+			createdSubscription := givenSubscription.DeepCopy()
+			// check if the created subscription is correct
+			getSubscriptionAssert(ctx, g, createdSubscription).Should(tc.wantSubscriptionMatchers)
+
+			// update subscription
+			givenUpdateSubscription.ResourceVersion = createdSubscription.ResourceVersion
+			ensureK8sResourceUpdated(ctx, t, givenUpdateSubscription)
+
+			// check if the updated subscription is correct
+			getSubscriptionAssert(ctx, g, givenSubscription).Should(tc.wantUpdateSubscriptionMatchers)
+		})
+	}
+}
+
+func Test_FixingSinkAndApiRule(t *testing.T) {
+	t.Parallel()
+	// common given test assets
+	givenSubscriptionFunc := func(namespace, name string) *eventingv1alpha2.Subscription {
```

Comment:
naming it `unreadySubscriptionWithoutSink` or something similar will give more clue of what you're trying to achieve here. Same for `givenUpdateSubscriptionFunc`, the `readySubscriptionWithSink` would also be better.
```suggestion
	unreadySubscriptionWithoutSink := func(namespace, name string) *eventingv1alpha2.Subscription {
```

Reply:
updated